registry/registry.go

```diff
@@ -237,8 +237,7 @@ func ContinueOnError(err error) bool {
 // default TLS configuration.
 func NewTransport(tlsConfig *tls.Config) *http.Transport {
 	if tlsConfig == nil {
-		var cfg = tlsconfig.ServerDefault
-		tlsConfig = &cfg
+		tlsConfig = &tlsconfig.ServerDefault
```

Comment:
Need someone to double check but I don't think these are the same things. I think the original will duplicate ServerDefault, while the new one just points to the default. Which means if someone (later on) modifies "tlsConfig" then they'll end up changing the default, which would be bad.